### DISCUSSION TITLE
How do you use useeffect filtering with incremental data loading (async iterable )?

### DISCUSSION THREAD

**Discussion by tiphaineruy** [visgl/deck.gl#6359] [Q&A]
When using  async iterable as per the example in the docs

```
// Create an async iterable
async function* getData() {
  for (let i = 0; i < 10; i++) {
    await const chunk = fetchChunk(...);
    yield chunk;
  }
}

function render() {
  const layer = new ScatterplotLayer({
    // When a new chunk arrives, deck.gl only updates the sub buffers for the new rows
    data: getData(),
    ...
  });

  deck.setProps({layers: [layer]});
}
```

How do you manage to  update / filter data while the data is being streamed ? Since you do not hold a reference to the DATA.

I'm thinking I should use the other streaming method and just create a layer with each data chunk streamed. But i'm not sure how having a few hundreds of layers instead of a single ref is a good thing.

**Comment by Pessimistress** [comment]
From the docs:

[data](https://deck.gl/docs/api-reference/core/layer#data)

> `AsyncIterable`: an async iterable object that yields data in batches. The default implementation expects each batch to be an array of data objects; one may change this behavior by supplying a custom `dataTransform` callback.

[dataTransform](https://deck.gl/docs/api-reference/core/layer#datatransform)
